src/gui/pedalboard/cables/CableView.cpp

```diff
@@ -31,45 +31,46 @@ void CableView::paint (Graphics& g)
         }
     }
 
-    g.setColour (cableColour.brighter (0.1f));
-    for (auto* cable : cables)
+    if (cableMouse != nullptr)
     {
-        auto* startEditor = board->findEditorForProcessor (cable->startProc);
-        jassert (startEditor != nullptr);
-
-        auto startPortLocation = CableViewPortLocationHelper::getPortLocation ({ startEditor, cable->startIdx, false });
-
-        if (cable->endProc != nullptr)
+        auto mousePos = cableMouse->getPosition();
+        const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cables.getLast()->startProc);
+        if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
         {
-            auto* endEditor = board->findEditorForProcessor (cable->endProc);
-            jassert (endEditor != nullptr);
+            auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
+            drawCablePortGlow (g, endPortLocation, scaleFactor);
+        }
+    }
+}
 
-            auto endPortLocation = CableViewPortLocationHelper::getPortLocation ({ endEditor, cable->endIdx, true });
+void CableView::resized()
+{
+    for (auto* cable : cables)
+    {
+        cable->setBounds (getLocalBounds());
+    }
+}
 
-            auto startColour = startEditor->getColour();
-            auto endColour = endEditor->getColour();
-            auto levelDB = cable->endProc->getInputLevelDB (cable->endIdx);
-            drawCable (g, startPortLocation.toFloat(), startColour, endPortLocation.toFloat(), endColour, scaleFactor, levelDB);
-        }
-        else if (connectionHelper->cableMouse != nullptr)
+void CableView::mouseDown (const MouseEvent& e)
+{
+    if (e.eventComponent == nullptr)
+        return; // not a valid mouse event
+
+    if (e.mods.isPopupMenu())
+    {
+        for (auto* cable : cables)
         {
-            auto mousePos = connectionHelper->cableMouse->getPosition();
-            const auto nearestInputPort = portLocationHelper->getNearestInputPort (mousePos, cable->startProc);
-            if (nearestInputPort.editor != nullptr && nearestInputPort.isInput && ! portLocationHelper->isInputPortConnected (nearestInputPort))
+            juce::Point clicked (e.getMouseDownX(), e.getMouseDownY());
+            if (cable->contains (clicked)) // Checks hitTest method for cable
             {
-                auto endPortLocation = CableViewPortLocationHelper::getPortLocation (nearestInputPort);
-                drawCablePortGlow (g, endPortLocation, scaleFactor);
+                board->popupMenu.showPopupMenu();
```

Comment:
Also, I think more recent JUCE's prefer `showMenuAsync()` for this kind of thing.